src/AstGen.zig

```diff
@@ -1718,6 +1718,61 @@ fn structInitExpr(
         }
     }
 
+    // Temporary Arena to simplify freeing the HashMaps
+    var dup_arena = std.heap.ArenaAllocator.init(astgen.arena);
+    const dup_alloc = dup_arena.allocator();
+
+    // Key: The first found variable.
+    // Value: A list of following duplicate variable names.
+    var duplicate_names = std.AutoHashMap(Ast.TokenIndex, std.ArrayList(Ast.TokenIndex)).init(dup_alloc);
+    try duplicate_names.ensureTotalCapacity(@intCast(struct_init.ast.fields.len));
+
+    // Basic check to maintain O(N) best case.
+    var isDuplicate = false;
```

Comment:
Follow the style conventions please - `snake_case` for non-callable variables. Also, I would name this `any_duplicate`.